include/seastar/http/client.hh

```diff
@@ -191,7 +191,7 @@ private:
     future<> shrink_connections();
 
     template <std::invocable<connection&> Fn>
-    auto with_connection(Fn&& fn, abort_source*);
+    futurize_t<std::invoke_result_t<Fn, connection&>> with_connection(Fn fn, abort_source*);
```

Comment:
heh, i believe Avi would like this change.

Reply:
It has less than 4 closing corner braces in a row, so not that bad

Meanwhile in scylla (scroll right):

```c++
test/boost/user_function_test.cc:769:        e.execute_cql("CREATE FUNCTION my_func(val int) CALLED ON NULL INPUT RETURNS map<int, frozen<set<frozen<list<frozen<tuple<text, bigint>>>>>>> LANGUAGE Lua AS  \
```

:scream:

Reply:
ah, i meant, he always prefers an explicit return type. this one make it more explicit. so..